From 0911aba1df0145ea468bab8d7a92facf0f88a67c Mon Sep 17 00:00:00 2001
From: Rob Evers <revers@redhat.com>
Date: Wed, 25 Jun 2014 14:11:43 -0400
Subject: [scsi] be2iscsi: fix bad if expression

Message-id: <1403705504-27660-10-git-send-email-revers@redhat.com>
Patchwork-id: 84805
O-Subject: [RHEL7.1 e-stor PATCH 09/10] scsi: be2iscsi: fix bad if expression
Bugzilla: 1088576
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Tomas Henzl <thenzl@redhat.com>
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>

From: Mike Christie <michaelc@cs.wisc.edu>

upstream scsi : 126e964a444f125bd428757fb88c24c730f6fcf9

https://bugzilla.kernel.org/show_bug.cgi?id=67091

Cc: Jayamohan Kallickal <Jayamohan.Kallickal@emulex.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/be2iscsi/be_main.c b/drivers/scsi/be2iscsi/be_main.c
index f2f034b..8cb49b0 100644
--- a/drivers/scsi/be2iscsi/be_main.c
+++ b/drivers/scsi/be2iscsi/be_main.c
@@ -329,7 +329,7 @@ static int beiscsi_eh_device_reset(struct scsi_cmnd *sc)
   if (!abrt_task->sc || abrt_task->state == ISCSI_TASK_FREE)
    continue;
 
-  if (abrt_task->sc->device->lun != abrt_task->sc->device->lun)
+  if (sc->device->lun != abrt_task->sc->device->lun)
    continue;
 
   /* Invalidate WRB Posted for this Task */
-- 
1.7.1